lib/bulk_insert/worker.rb

```diff
@@ -64,34 +68,50 @@ def after_save(&block)
 
     def save!
       if pending?
-        sql = "INSERT #{@ignore} INTO #{@table_name} (#{@column_names}) VALUES "
-        @now = Time.now
-
-        rows = []
-        @set.each do |row|
-          values = []
-          @columns.zip(row) do |column, value|
-            value = @now if value == :__timestamp_placeholder
-
-            if ActiveRecord::VERSION::STRING >= "5.0.0"
-              value = @connection.type_cast_from_column(column, value) if column
-              values << @connection.quote(value)
-            else
-              values << @connection.quote(value, column)
-            end
-          end
-          rows << "(#{values.join(',')})"
-        end
-
-        sql << rows.join(",")
-        @connection.execute(sql)
-
+        @connection.execute(compose_insert_query)
         @after_save_callback.() if @after_save_callback
-
         @set.clear
       end
 
       self
     end
+
+    private
```

Comment:
I'm not a fan of Ruby's visibility controls. Let's leave them off, and prefix internal method names with an underscore. (e.g., `def _compose_insert_query`).